statics/js/components/rule_detail.js

```diff
@@ -0,0 +1,489 @@
+   */
+  BridgeLayout.prototype.switch_port_tab = function (tabR, tabP) {
+    $('.nav-tabs a[href="#P' + tabR + '-' + tabP + '"]').tab('show');
+  }
+
+  /** hightlights the node associated to a port
+   * @param p: Openflow index of the port to hightlight.
+   */
+  BridgeLayout.prototype.mark = function(p) {
+    var nodes = this.interfaces[p];
+    for (var i = 0; i < nodes.length; i++) {
+      this.store.commit('highlight', nodes[i].id);
+    }
+  };
+
+  /** unhightlights the node associated to a port
+   * @param p: index of the port to unhighlight
+   */
+  BridgeLayout.prototype.unmark = function(p) {
+    var nodes = this.interfaces[p];
+    for (var i = 0; i < nodes.length; i++) {
+      this.store.commit('unhighlight', nodes[i].id);
+    }
+  };
+
+  /** Select the node at the output of a rule.
+     * @param p: the index of the port to follow to find the new selected node.
+     */
+  BridgeLayout.prototype.select = function(p) {
+    var nodes = this.interfaces[p];
+    this.unmark(p);
+    var len = nodes.length;
+    var last = nodes[len - 1];
+    if (len === 4) {
+      last = this.graph.getNeighbor(last, 'ovsbridge');
+    }
+    this.store.commit('selected', last);
+    this.switch_tab(0);
+  };
+
+  /** Path to the action icon
+   * @param the action to represent
+   * @return path of the icon to use.
+  */
+  BridgeLayout.prototype.imgsrc = function(act) {
+    var file;
+    switch (act) {
+      case ActionOutput:
+        file = 'output.png';
+        break;
+      case ActionResubmit:
+        file = 'resubmit.png';
+        break;
+      case ActionFlood:
+        file = 'flood.png';
+        break;
+      case ActionNormal:
+        file = 'flood.png';
+        break;
+      case ActionDrop:
+        file = 'drop.png';
+      default:
+        return null;
+    }
+    return 'statics/img/openflow/' + file;
+  };
+
+  /** Readable name of an openflow port
+   * @param pr the index of the port or its stringified value.
+  */
+  BridgeLayout.prototype.portname = function(pr) {
+    var p = (typeof (pr) === 'string') ? parseInt(pr) : pr;
+    if (p === ANY_PORT) return 'ANY';
+    if (p === LOCAL_PORT) return 'LOCAL';
+    if (p === SAME_PORT || p === undefined) return '';
+    var nodes = this.interfaces[p];
+    if (nodes === undefined) return '???';
+    var port = nodes[1];
+    var portname = (port === undefined) ? '???' : port.metadata.Name;
+    return portname;
+  };
+
+  /** Check if rule is highlighted. */
+  BridgeLayout.prototype.isHighlighted = function(rule) {
+    var current = this.store.state.currentRule;
+    var status = current && rule.UUID === current.metadata.UUID;
+    return status;
+  };
+
+  return BridgeLayout;
+}());
+
+/** Graphical component that represents a set of rules associated to a given
+ *  Openflow table and port and displayed as a single HTML table.
+ */
+Vue.component('rule-table-detail', {
+  template: '\
+              <table class="table table-bordered table-condensed">\
```

Comment:
I have adapted the look but not reused dynamic-table without documentation. I am a little afraid of using it on the summary field.